Fix path resolution for server files and add a CHANGELOG.md

parkervcp opened the pull request:
This is to duplicate a change to the main daemon.

Also fixed permissions handling on the golang sftp daemon.

parkervcp committed:
ignore sftp-server.log

parkervcp committed:
Ensure correct server data path is used

This is to duplicate a change to the main daemon.

parkervcp committed:
updated file handler

Fixed file perms when uploading and changing file permissions.

Fixed path resolution for server files

parkervcp committed:
Add a CHANGELOG.md

parkervcp commented:
Just a note that some of the re-organization in the imports it due to VSCode and their formatting handling.

DaneEveritt commented on CHANGELOG.md:
```diff
@@ -0,0 +1,13 @@
+
+## v1.0.3
+
+### New Feature
```
Use `Added`, `Removed`, `Fixed` here, and the security one can just be `[Security]` in front of the text under Fixed.

parkervcp replied:
This has been resolved.

parkervcp committed:
code cleanup and real perms fix

Moved path check code to unified location.
Corrected the file perms issue for real this time defaults to 0644 but can now be edited.

parkervcp committed:
Order of operations is important.

DaneEveritt commented on src/server/handler.go:
```diff
@@ -180,7 +181,14 @@ func (fs FileSystem) Filecmd(request *sftp.Request) error {
+		if request.Attributes().FileMode().Perm() != 0000 {
+			mode = request.Attributes().FileMode().Perm()
+		} else {
+			mode = 0644
```
Nitpick, but can you do `var mode os.FileMode = 0644;` and just have the if with no else here?

parkervcp replied:
No. This makes it so you can change the perms after it's been uploaded.

The other way made it so you couldn't update perms.

DaneEveritt replied:
I mean just remove the need for the else block.

DaneEveritt commented on src/server/handler.go:
```diff
@@ -272,7 +280,7 @@ func (fs FileSystem) Filecmd(request *sftp.Request) error {
 }
 
-// Handler for SFTP filesystem list calls. This will handle calls to list the contents of
+// Filelist is the andler for SFTP filesystem list calls. This will handle calls to list the contents of
```
typo

parkervcp replied:
This is resolved.

DaneEveritt commented on src/server/handler.go:
```diff
@@ -331,6 +339,23 @@ func (fs FileSystem) buildPath(rawPath string) (string, error) {
+	// Check if the path is in the server directory and return a no if it isn't.
+	symfile, err := filepath.EvalSymlinks(p)
+	if err != nil {
+		return "", errors.New("Error Evaluating Symlink Path")
```
error string should be all lowercase.

parkervcp replied:
Fixed this.

parkervcp committed:
Correcting errors.

DaneEveritt committed:
Little bit of code cleanup and clarifying comments
